internal/resolution/variablesources/bundle_deployment.go

```diff
@@ -34,21 +32,18 @@ func (o *BundleDeploymentVariableSource) GetVariables(ctx context.Context) ([]de
 		variableSources = append(variableSources, o.inputVariableSource)
 	}
 
-	processed := sets.Set[string]{}
-	for _, bundleDeployment := range o.bundleDeployments {
-		sourceImage := bundleDeployment.Spec.Template.Spec.Source.Image
-		if sourceImage != nil && sourceImage.Ref != "" {
-			if processed.Has(sourceImage.Ref) {
-				continue
-			}
-			processed.Insert(sourceImage.Ref)
-			ips, err := NewInstalledPackageVariableSource(o.allBundles, bundleDeployment.Spec.Template.Spec.Source.Image.Ref)
-			if err != nil {
-				return nil, err
-			}
-			variableSources = append(variableSources, ips)
-		}
+	variables, err := variableSources.GetVariables(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	installedPackages, err := MakeInstalledPackageVariables(o.allBundles, o.bundleDeployments)
+	if err != nil {
+		return nil, err
```

Comment:
Would it be beneficial to wrap these errors with more context before returning?

Reply:
I think on this level we should just pass them down the call stack. In `MakeInstalledPackageVariables` we should be able to add a bit more context (e.g. specific bundle deployment name).

Reply:
@everettraven I'm going to resolve this thread. Please re-open if you think we should change something here.